scripts-dev/lint.sh

```diff
@@ -7,15 +7,88 @@
 
 set -e
 
-if [ $# -ge 1 ]
-then
-  files=$*
+usage() {
+  echo
+  echo "Usage: $0 [-h] [-d] [paths...]"
+  echo
+  echo "-d"
+  echo "  Lint files that have changed since the last git commit."
+  echo
+  echo "  If paths are provided and this option is set, both provided paths and those"
+  echo "  that have changed since the last commit will be linted."
+  echo
+  echo "  If no paths are provided and this option is not set, all files will be linted."
+  echo
+  echo "  Note that paths with a file extension that is not '.py' will be excluded."
+  echo "-h"
+  echo "  Display this help text."
+}
+
+USING_DIFF=0
+files=()
+
+while getopts ":dh" opt; do
+  case $opt in
+    d)
+      USING_DIFF=1
+
+      # Check both staged and non-staged changes
+      for path in $(git diff HEAD --name-only); do
+        filename=$(basename "$path")
+        file_extension="${filename##*.}"
+
+        # If an extension is present, and it's something other than 'py',
+        # then ignore this file
+        if [[ -n ${file_extension+x} && $file_extension != "py" ]]; then
+          continue
+        fi
+
+        # Append this path to our list of files to lint
+        files+=("$path")
+      done
+      ;;
+    h)
+      usage
+      exit
+      ;;
+    \?)
+      echo "ERROR: Invalid option: -$OPTARG" >&2
+      usage
+      exit
+      ;;
+  esac
+done
+
+# Strip any options from the command line arguments now that
+# we've finished processing them
+shift "$((OPTIND-1))"
+
+# Append any remaining arguments as files to lint
+files+=("$@")
```

Comment:
it feels to me like trying to mix `-d` and command-line parameters should be rejected: mixing the two seems confusing.

Reply:
I suppose there's not much use-case for it... though I'm not sure if we explicitly need to block it either.